budget_control/models/analytic_account.py

```diff
@@ -205,11 +205,13 @@ def _check_budget_control_status(self, budget_period_id=False):
                 _("Following analytics has no budget control sheet:\n%s")
                 % ", ".join(names)
             )
-        budget_not_controlled = budget_controls.filtered_domain(
-            [("state", "!=", "done")]
-        )
-        if budget_not_controlled:
-            names = budget_not_controlled.mapped("analytic_account_id.display_name")
+        # Find analytics has no controlled budget control sheet
+        budget_controlled = budget_controls.filtered_domain([("state", "=", "done")])
+        cbc_analytic_ids = budget_controlled.mapped("analytic_account_id").ids
+        no_cbc_analytic_ids = list(set(self.ids) - set(cbc_analytic_ids))
```

Comment:
Can we oper with objects directly? So you don't need to browse again. For better performance.

Reply:
Thank you, I fixed it.